torch/ao/quantization/fx/_lower_to_native_backend.py

```diff
@@ -63,6 +73,77 @@ def is_fixed_qparams_node(node, modules):
     nni.LinearReLU: (nnqr.Linear, nniq.LinearReLU)
 }
 
+# Mapping from a functional to lower to a 2-tuple of
+#   1) The quantized version of the op
+#   2) The quantized version of the op fused with relu, if it exists, else None
+LOWER_FUNCTIONAL_MAP = {
+    F.linear: (torch.ops.quantized.linear, torch.ops.quantized.linear_relu),
+}
+
+WEIGHT_PREPACK_OPS = {
+    torch._ops.ops.quantized.linear_prepack,
+    torch._ops.ops.quantized.linear_prepack_fp16,
+    torch._ops.ops.quantized.conv1d_prepack,
+    torch._ops.ops.quantized.conv2d_prepack,
+    torch._ops.ops.quantized.conv3d_prepack,
+}
+
+def fold_weight(
+        quantized: QuantizedGraphModule,
+        node_name_to_scope: Dict[str, Tuple[str, type]]) -> QuantizedGraphModule:
+    """
+    Trace back from the weight node util we hit getattr, reconstruct the
+    graph module with the traced nodes and run the graph module to pack the
+    weight. then replace the original chain of ops with the packed weight.
+    """
+    packed_weights = dict()
+    # map from folded node name to the prepacked weight name
+    folded_nodes = dict()
+    # get packed weights
+    for node in quantized.graph.nodes:
+        if node.op == 'call_function' and node.target in WEIGHT_PREPACK_OPS:
+            nodes_to_fold = collect_producer_nodes(node)
+            if nodes_to_fold is not None:
+                for node_to_fold in nodes_to_fold:
+                    folded_nodes[node_to_fold.name] = node
+
+                prepacking_module = graph_module_from_producer_nodes(
+                    quantized, nodes_to_fold)
+                packed_weight = prepacking_module()
+                packed_weights[node.name] = packed_weight
+
+    # remove folded nodes and replace the prepacking node with getattr
+    folded_graph = Graph()
+    env: Dict[Any, Any] = {}
+
+    def load_arg(a):
+        return map_arg(a, lambda node: env[node.name])
+    quantized_root = quantized
+    quantized_graph = quantized.graph
+
+    for node in quantized_graph.nodes:
+        prepack_node = folded_nodes.get(node.name, None)
+        if prepack_node is node:
+            packed_weight = packed_weights[node.name]
+            # add a prepacked attribute to root
+            op_node = list(prepack_node.users)[0]
+            module_path, _ = node_name_to_scope.get(op_node.name, ('', None))
```

Comment:
This function was moved from `fx/convert.py` and this was the only line I changed. Previously it was
```
module_path, _ = node_name_to_scope.get[op_node.name]
```
but this gave me a key error for `linear_relu` so I added a default